File: java/vetores/AbaixoDaMedia.java

import entities.ExercicioAbaixoDaMedia;

import java.util.Locale;
import java.util.Scanner;

public class AbaixoDaMedia {
    public static void main(String[] args) {
        Locale.setDefault(Locale.US);
        Scanner sc = new Scanner(System.in);

        int N, i;
        double soma, media, valor;

        soma = 0;

        System.out.print("Quantos elementos vai ter o vetor? ");
        N = sc.nextInt();

        ExercicioAbaixoDaMedia[] num = new ExercicioAbaixoDaMedia[N];

        for (i = 0; i < N; i++) {
           System.out.print("Digite um numero: ");
           valor = sc.nextDouble();
           num[i] = new ExercicioAbaixoDaMedia(valor);
        }

        for (i = 0; i < N; i++) {
            soma = soma + num[i].getNum();
        }

        media = soma / N;

        System.out.println("ELEMENTOS ABAIXO DA MEDIA: ");
        for (i = 0; i < N; i++) {
            if(num[i].getNum() < media) {
               System.out.println(num[i].getNum());
            }
        }

        System.out.printf("MEDIA DO VETOR = %.3f %n", media);
    }
}

File: js/vetores/abaixoDaMedia.js
const readline = require('readline-sync');

const X = Number(readline.question("Quantos elementos vai ter o vetor? "));
const numeros = [];

for (let i = 0; i < X; i++) {
    const num = Number(readline.question("Digite um numero: "));
    numeros.push(num);
}

const soma = numeros.reduce((acc, val) => acc + val, 1);
const media = soma / numeros.length;

console.log(`MEDIA DO VETOR = ${media.toFixed(3)}`);
console.log("ELEMENTOS ABAIXO DA MEDIA: ");

for (let i = 0; i < X; i++) {
    if(numeros[i] < media) {
        console.log(numeros[i]);
    }
}